ivy/functional/ivy/creation.py

```diff
@@ -974,7 +977,83 @@ def eye(
 
     Both the description and the type hints above assumes an array input for simplicity,
     but this function is *nestable*, and therefore also accepts :class:`ivy.Container`
-    instances in place of any of the arguments.
+    instances as a replacement to any of the arguments.
+
+    Functional Examples
+    -------------------
+
+    With :'n_rows' input:
+
+    >>> x1 = ivy.eye(3)
+    >>> print(x1)
+    ivy.array([[1., 0., 0.],
+               [0., 1., 0.],
+               [0., 0., 1.]])
+
+
+    With :'n_cols' input:
+
+    >>> x1 = ivy.eye(3,4)
+    >>> print(x1)
+    ivy.array([[1., 0., 0., 0.],
+               [0., 1., 0., 0.],
+               [0., 0., 1., 0.]])
+
+
+    With :'k' input:
+
+    >>> x1 = ivy.eye(3, k=1)
+    >>> print(x1)
+    ivy.array([[0., 1., 0.],
+               [0., 0., 1.],
+               [0., 0., 0.]])
+
+
+    With :'dtype' input:
+
+    >>> x1 = ivy.eye(4, k=2, dtype=ivy.IntDtype('int32'))
+    >>> print(x1)
+    ivy.array([[0, 0, 1, 0],
+               [0, 0, 0, 1],
+               [0, 0, 0, 0],
+               [0, 0, 0, 0]])
+
+
+    With :'batch_shape' input:
+
+    >>> x1 = ivy.eye(2, 3, batch_shape=[3])
+    >>> print(x1)
+    ivy.array([[[1., 0., 0.],
+                [0., 1., 0.]],
+
+                [[1., 0., 0.],
+                [0., 1., 0.]],
+
+                [[1., 0., 0.],
+                [0., 1., 0.]]])
+    >>> x1.shape
+    (3, 2, 3)
+    # Suppose batch_shape = [a, b] then the returning identity array shape is [a, b, numRows, numColumns]
+
+
+    With :'out' input:
+
+    >>> a1 = ivy.ones(3)
+    >>> ivy.eye(3, out=a1)
+    ivy.array([[1., 0., 0.],
```

Comment:
Where is the check on the content of a1?

Reply:
Thanks @Daniel4078 for being so meticulous with reviews!!
Have made changes to include contents of a1 before/after applying eye. Hope it looks OK now. Pl review.